query_plan.go

```diff
@@ -104,15 +105,48 @@ func getMetadataString(node *Node, key string) (string, bool) {
 	}
 }
 
-func getAllMetadataString(node *Node) string {
-	if node.PlanNode.Metadata == nil {
-		return ""
+func getNodeTitle(node *Node) string {
+	fields := node.PlanNode.GetMetadata().GetFields()
+
+	var components []string
+	for _, s := range []string{
+		fields["call_type"].GetStringValue(),
+		fields["iterator_type"].GetStringValue(),
+		strings.TrimSuffix(fields["scan_type"].GetStringValue(), "Scan"),
+		node.PlanNode.GetDisplayName(),
+	} {
+		if s != "" {
+			components = append(components, s)
+		}
 	}
+	return strings.Join(components, " ")
+}
+
+func getAllMetadataString(node *Node) string {
+	metadata := node.PlanNode.GetMetadata().GetFields()
 
 	fields := make([]string, 0)
-	for k, v := range node.PlanNode.Metadata.Fields {
-		fields = append(fields, fmt.Sprintf("%s: %s", k, v.GetStringValue()))
+	for k, v := range metadata {
+		switch k {
+		case "call_type", "iterator_type": // Skip because it is displayed in node title
```

Comment:
It looks like implementations of getNodeTitle() and getAllMetadataString() are coupled with each other.
(getAllMetadataString() depends on the implementation of getNodeTitle()).

How about merging them into a single function?
```
func (n *Node) String() string {
  // assemble operator
  ...
  // assemble metadata
  ...
  return operator + metadata
}
```

Reply:
Fixed in https://github.com/cloudspannerecosystem/spanner-cli/pull/63/commits/8d7a1b4193141c1dbd0590472b2525f9b4dc1582